pkg/cmd/cli/cmd/set/volume.go

```diff
@@ -609,6 +609,10 @@ func (v *VolumeOptions) addVolumeToSpec(spec *kapi.PodSpec, info *resource.Info,
 			}
 			spec.Volumes = append(spec.Volumes[:i], spec.Volumes[i+1:]...)
 			break
+		} else {
+			if opts.Overwrite {
```

Comment:
So thinking about this more, generally we don't treat `--force` or `--confirm` as negative errors (if you say --force to something that doesn't require forcing, we don't error out).  What was the user case that you hit that resulted in you wanting to change this?  Did it result in something broken?